src/api/routes/v2/addresses.ts

```diff
@@ -118,5 +125,86 @@ export const AddressRoutesV2: FastifyPluginAsync<
     }
   );
 
+  // get balances for STX, FTs, and counts for NFTs
+  fastify.get(
+    '/:principal/balances',
+    {
+      preHandler: handlePrincipalMempoolCache,
+      schema: {
+        operationId: 'get_principal_balances',
+        summary: 'Get principal balances',
+        description: `Retrieves total account balance information for a given Address or Contract Identifier. This includes the balances of STX Tokens and Fungible Tokens for the account.`,
+        tags: ['Accounts'],
+        params: Type.Object({
+          principal: PrincipalSchema,
+        }),
+        response: {
+          200: AddressBalanceV2Schema,
+        },
+      },
+    },
+    async (req, reply) => {
+      const stxAddress = req.params.principal;
+      validatePrincipal(stxAddress);
+
+      const result = await fastify.db.sqlTransaction(async sql => {
+        const chainTip = await fastify.db.getChainTip(sql);
+
+        // Get balances for fungible tokens
+        const ftBalancesResult = await fastify.db.v2.getFungibleTokenHolderBalances({
```

Comment:
I think this is a good opportunity to paginate this list somehow... I know there are some addresses that have hundreds of tokens (DEX contracts and such).

Maybe a good approach would be to move these to a new endpoint? `/:principal/balances/ft` (similar to the NFT holdings endpoint). Paginating a nested object in this endpoint would be kind of weird imo

Reply:
Good call, I'll work on that

Reply:
Done